### PR TITLE
[MLDBFB-618] Calling {row|column}PathElement() with an index bigger t…

### PR DESCRIPTION
**Pull request by mailletf**
…han the length of the path will simply return NULL instead of throwing

### CODE REVIEW

File: sql/binding_contexts.cc [hunk 1]
```diff
@@ -589,15 +589,13 @@ getDatasetDerivedFunction(const Utf8String & tableName,
...
                         firstElement = asPath.size() + firstElement;
-                    if (firstElement < 0 || firstElement >= asPath.size()) {
-                        throw HttpReturnException
-                            (400, "Couldn't extract element '"
-                             + to_string(firstElement) + "' of path '"
-                             + asPath.toUtf8String() + "' with length "
-                             + to_string(asPath.size()));
-                    }
+
+                    if (firstElement < 0 || firstElement >= asPath.size())
```
Comment (jeremybarnes) [inline sql/binding_contexts.cc]: We could consider allowing negative numbers to index from the end as in Python.

**Comment by jeremybarnes** [comment]
It needs some kind of mention in the documentation.

File: sql/binding_contexts.cc (hunk 1 above)
Comment (mailletf) [inline sql/binding_contexts.cc]: Adding a test for it and it already works

**Comment by mailletf** [comment]
tests qui fail ont pas rapport

**Comment by FinchPowers** [comment]
+1, reroule ton test, Jenkins devrait être réparé.

**Pull request merged**